pkg/bigint/bigint.go

```diff
@@ -0,0 +1,37 @@
+package bigint
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+)
+
+type BigInt struct {
+	big.Int
```

Comment:
should be `*big.Int`. big ints are passed around as pointers, this has all sorts of implications but your type should probably wrap the pointer instead of an instance